libbpf-rs/src/object.rs

```diff
@@ -167,12 +165,58 @@ pub struct OpenObject {
 }
 
 impl OpenObject {
-    fn new(ptr: *mut libbpf_sys::bpf_object) -> Self {
-        OpenObject {
+    fn new(ptr: *mut libbpf_sys::bpf_object) -> Result<Self> {
+        let mut obj = OpenObject {
             ptr,
             maps: HashMap::new(),
             progs: HashMap::new(),
+        };
+
+        // Populate obj.maps
+        let mut map: *mut libbpf_sys::bpf_map = std::ptr::null_mut();
+        loop {
+            // Get the pointer to the next BPF map
+            let next_ptr = unsafe { libbpf_sys::bpf_map__next(map, obj.ptr) };
+            if next_ptr.is_null() {
+                break;
+            }
+
+            // Get the map name
+            let name = unsafe { libbpf_sys::bpf_map__name(next_ptr) };
```

Comment:
bpf_map__name never returns NULL if you pass non-NULL map pointer. Even when you pass NULL it returns NULL, so there is no need to use libbpf_get_error. So please just simplify all this.

Reply:
Fixed in latest commit